llvm/lib/Transforms/Scalar/StructurizeCFG.cpp

```diff
@@ -256,6 +256,10 @@ class StructurizeCFG {
   BBPhiMap DeletedPhis;
   BB2BBVecMap AddedPhis;
 
+  SmallVector<BBValueVector> BBValuesPool;
+  // Map PHINode to the index of the merged incoming values in BBValuesPool
+  DenseMap<PHINode *, unsigned> MergedPHIMap;
```

Comment:
Can you use EquivalenceClasses for this? It seems like you're doing something similar

Reply:
Yeah, it helps simplify the implementation, especially on handling the case that two phi nodes are already merged with other phi nodes.